tests/test_option_model.py

```diff
@@ -18,32 +26,46 @@ def test_default_option_model():
     obj4 = type2("obj4")
     obj9 = type2("obj9")
 
-    def _simulate(state, action):
-        next_state = state.copy()
-        obj = list(state)[0]
-        next_state.set(obj, "feat3",
-                       next_state.get(obj, "feat3") + action.arr[1])
-        return next_state
+    class _MockEnv:
+
+        @staticmethod
+        def simulate(state: State, action: Action) -> State:
+            """A mock simulate method."""
+            next_state = state.copy()
+            obj = list(state)[0]
+            next_state.set(obj, "feat3",
+                           next_state.get(obj, "feat3") + action.arr[1])
+            return next_state
 
-    params_space = Box(-10, 10, (2, ))
+        @property
+        def options(self) -> Set[ParameterizedOption]:
+            """Mock options."""
+            params_space = Box(-10, 10, (2, ))
 
-    def policy(s, m, o, p):
-        del s, m, o  # unused
-        return Action(p * 2)
+            def policy(s, m, o, p):
+                del s, m, o  # unused
+                return Action(p * 2)
 
-    def initiable(s, m, o, p):
-        del o, p  # unused
-        obj = list(s)[0]
-        m["start_state"] = s
-        return s[obj][0] < 10 or s[obj][0] > 60
+            def initiable(s, m, o, p):
+                del o, p  # unused
+                obj = list(s)[0]
+                m["start_state"] = s
+                return s[obj][0] < 10 or s[obj][0] > 60
 
-    def terminal(s, m, o, p):
-        del o, p  # unused
-        obj = list(s)[0]
-        return s[obj][0] > 50 and not s.allclose(m["start_state"])
+            def terminal(s, m, o, p):
+                del o, p  # unused
+                obj = list(s)[0]
+                return s[obj][0] > 50 and not s.allclose(m["start_state"])
+
+            parameterized_option = ParameterizedOption("Pick", [],
+                                                       params_space, policy,
+                                                       initiable, terminal)
+
+            return {parameterized_option}
+
+    env = _MockEnv()
+    parameterized_option = env.options.pop()
```

Comment:
how come this pop doesn't mess up the env's options? is it because `env.options` creates a new set object every time?

Reply:
yeah